src/plugins/no_data_page/public/lib/use_has_api_keys.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { HttpSetup } from '@kbn/core-http-browser';
+import { useEffect, useState } from 'react';
+import { NO_DATA_API_PATHS } from '../../common';
+import { HasApiKeysResponse } from '../types';
+
+export const createUseHasApiKeys = ({ http }: { http: HttpSetup }) => {
+  return function useHasApiKeys(): HasApiKeysResponse {
+    const [hasApiKeys, setHasApiKeys] = useState<boolean | null>(null);
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState<Error | null>(null); // NOTE: error is NOT re-thrown
+
+    useEffect(() => {
+      setLoading(true);
+      http
+        .get<HasApiKeysResponse>(NO_DATA_API_PATHS.internal.hasApiKeys)
```

Comment:
@legrego @azasypkin The purpose of this React hook is to utilize the Security API endpoint being added in: https://github.com/elastic/kibana/pull/172884. It's exposed in the NoDataPage plugin as a stateful service, having the `http` dependency baked in. Would it be acceptable to move this hook into the Security domain? Either as a stateful service like it is here, or part of a package (in which case the consumer must pass the `http` dependency). Thoughts?

Reply:
Currently, with only one usage, I don't have a strong preference. If we had multiple plugins requiring this hook, I'd advocate for moving it to the Security plugin. Until then, my suggestion would be to potentially shift only the part that calls the Security internal endpoint. This way, you won't have to worry if we decide to alter the URL or response format in the future (e.g., `securityPlugin.start.authc.apiKeys.hasAnyKeys(): Promise<boolean>` or a similar approach). But up to you.

Reply:
We're going to have multiple plugins requiring it very soon :)

Reply:
Aha, good to know! Then it definitely makes sense to have it as part of the Security area. It's up to you — either move the hook now or when you start using it in the second plugin. I slightly prefer the latter as plans can change, but I'm fine either way.

> Either as a stateful service in a plugin contract, like it is here, or part of a package (in which case the consumer must pass the http dependency). 

What approach would work better for you as a consumer? I assume you'll also need an easy way to mock HTTP responses (or some other way to force hook to return different values) in tests to evaluate various scenarios in UI tests?

Reply:
Thanks @azasypkin I am fine with keeping it as part of the `no_data_page` plugin for now. If we see a need to use this hook outside of this plugin, we can move it to a package. A package would work fine here. The hook can be easily tested with utilities in the `@testing-library/react-hooks` package. :)